### PR TITLE
[2.0.x] Fixed #322

### LINKED ISSUE #322: [2.0.x] View\Engine compatlibity
**\Phalcon\Mvc\View\Engine\Twig::construct()** must be compatible with **\Phalcon\Mvc\View\EngineInterface::__construct($view, \Phalcon\DiInterface $dependencyInjector = null)**

\Phalcon\Mvc\View\Engine\Twig::__construct:
```php
 /**
     * {@inheritdoc}
     *
     * @param \Phalcon\Mvc\ViewInterface $view
     * @param \Phalcon\DiInterface       $dependencyInjector
     */
    public function __construct($view, \Phalcon\DiInterface $dependencyInjector = null)
    {
        $loader     = new \Twig_Loader_Filesystem($view->getViewsDir());
        $this->twig = new Twig\Environment($dependencyInjector, $loader);

        $this->twig->addExtension(new Twig\CoreExtension());
        $this->registryFunctions($view, $dependencyInjector);

        parent::__construct($view, $dependencyInjector);
    }
```

### PR DESCRIPTION
**Pull request by CawaKharkov**
Fix for Phaclon 2.0 new EngineInterface
```php
public function __construct($view, \Phalcon\DiInterface $dependencyInjector = null);
```

### CODE REVIEW

File: Library/Phalcon/Mvc/View/Engine/Smarty.php [hunk 1]
```diff
@@ -23,7 +23,7 @@ class Smarty extends Engine implements EngineInterface
      * @param \Phalcon\Mvc\ViewInterface $view
      * @param \Phalcon\DiInterface       $di
      */
-    public function __construct($view, DiInterface $di = null)
+    public function __construct($view, \Phalcon\DiInterface $dependencyInjector = null)
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/View/Engine/Smarty.php]: aliasing is accomplished with the use operator, see above:
```php
use Phalcon\DiInterface;
```

File: Library/Phalcon/Mvc/View/Engine/Twig.php [hunk 2]
```diff
@@ -21,27 +21,25 @@ class Twig extends Engine implements EngineInterface
      * {@inheritdoc}
      *
      * @param \Phalcon\Mvc\ViewInterface $view
-     * @param \Phalcon\DiInterface       $di
-     * @param array                      $options
-     * @param array                      $userFunctions
+     * @param \Phalcon\DiInterface $dependencyInjector
      */
-    public function __construct($view, DiInterface $di = null, $options = array(), $userFunctions = array())
+    public function __construct($view, \Phalcon\DiInterface $dependencyInjector = null)
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/View/Engine/Twig.php]: `\Phalcon\DiInterface` → `DiInterface`

File: Library/Phalcon/Mvc/View/Engine/Twig.php [hunk 3]
```diff
@@ -21,27 +22,25 @@ class Twig extends Engine implements EngineInterface
...
+        parent::__construct($view, $dependencyInjector);
     }
 
     /**
      * Registers common function in Twig
      *
      * @param \Phalcon\Mvc\ViewInterface $view
-     * @param \Phalcon\DiInterface       $di
-     * @param array                      $userFunctions
+     * @param \Phalcon\DiInterface $di
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/View/Engine/Twig.php]: Return back indents please

File: Library/Phalcon/Mvc/View/Engine/Twig.php [hunk 4]
```diff
@@ -136,8 +135,8 @@ protected function registryFunctions($view, DiInterface $di, $userFunctions = ar
     /**
      * {@inheritdoc}
      *
-     * @param string  $path
-     * @param array   $params
+     * @param string $path
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/View/Engine/Twig.php]: Return back indents please

File: Library/Phalcon/Mvc/View/Engine/Twig.php [hunk 5]
```diff
@@ -1,9 +1,10 @@
 <?php
+
 namespace Phalcon\Mvc\View\Engine;
 
+use Phalcon\DiInterface;
 use Phalcon\Mvc\View\Engine;
 use Phalcon\Mvc\View\EngineInterface;
-use Phalcon\DiInterface;
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/View/Engine/Twig.php]: Return this back please

File: Library/Phalcon/Mvc/View/Engine/Twig.php [hunk 6]
```diff
@@ -21,27 +22,25 @@ class Twig extends Engine implements EngineInterface
      * {@inheritdoc}
      *
      * @param \Phalcon\Mvc\ViewInterface $view
-     * @param \Phalcon\DiInterface       $di
-     * @param array                      $options
-     * @param array                      $userFunctions
+     * @param \Phalcon\DiInterface $dependencyInjector
      */
-    public function __construct($view, DiInterface $di = null, $options = array(), $userFunctions = array())
+    public function __construct($view, DiInterface $dependencyInjector = null)
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/View/Engine/Twig.php]: Return `options` and `userFunctions` back please.

See http://stackoverflow.com/questions/15413829/why-does-php-allow-incompatible-constructors

File: Library/Phalcon/Mvc/View/Engine/Smarty.php [hunk 7]
```diff
@@ -21,26 +22,26 @@ class Smarty extends Engine implements EngineInterface
...
     {
-        $this->smarty               = new \Smarty();
+        $this->smarty = new \Smarty();
         $this->smarty->template_dir = '.';
-        $this->smarty->compile_dir  = SMARTY_DIR . 'templates_c';
-        $this->smarty->config_dir   = SMARTY_DIR . 'configs';
-        $this->smarty->cache_dir    = SMARTY_DIR . 'cache';
-        $this->smarty->caching      = false;
-        $this->smarty->debugging    = true;
+        $this->smarty->compile_dir = SMARTY_DIR . 'templates_c';
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/View/Engine/Smarty.php]: Return indents back please

File: Library/Phalcon/Mvc/View/Engine/Twig.php [hunk 8]
```diff
@@ -21,17 +22,17 @@ class Twig extends Engine implements EngineInterface
      * {@inheritdoc}
      *
      * @param \Phalcon\Mvc\ViewInterface $view
-     * @param \Phalcon\DiInterface       $di
-     * @param array                      $options
-     * @param array                      $userFunctions
+     * @param \Phalcon\DiInterface $dependencyInjector
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/View/Engine/Twig.php]:
```php
@param \Phalcon\DiInterface $dependencyInjector
public function __construct($view, DiInterface $di = null, $options = array(), $userFunctions = array())
```

and here

File: Library/Phalcon/Mvc/View/Engine/Smarty.php [hunk 9]
```diff
@@ -21,26 +22,26 @@ class Smarty extends Engine implements EngineInterface
      * {@inheritdoc}
      *
      * @param \Phalcon\Mvc\ViewInterface $view
-     * @param \Phalcon\DiInterface       $di
+     * @param \Phalcon\DiInterface $di
      */
-    public function __construct($view, DiInterface $di = null)
+    public function __construct($view, DiInterface $dependencyInjector = null)
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/View/Engine/Smarty.php]:
```php
 @param \Phalcon\DiInterface $di

public function __construct($view, DiInterface $dependencyInjector = null)
```

$di or $dependencyInjector ?
For what purpose you rename it? What are the benefits?

return back `$di` please

File: Library/Phalcon/Mvc/View/Engine/Twig.php [hunk 10]
```diff
@@ -21,17 +22,17 @@ class Twig extends Engine implements EngineInterface
...
+     * @param array $options
+     * @param array $userFunctions
      */
     public function __construct($view, DiInterface $di = null, $options = array(), $userFunctions = array())
     {
-        $loader     = new \Twig_Loader_Filesystem($view->getViewsDir());
-        $this->twig = new Twig\Environment($di, $loader, $options);
+        $loader = new \Twig_Loader_Filesystem($view->getViewsDir());
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/View/Engine/Twig.php]: return back $options please

File: Library/Phalcon/Mvc/View/Engine/Twig.php [hunk 11]
```diff
@@ -21,17 +22,17 @@ class Twig extends Engine implements EngineInterface
...
+        $this->twig = new Twig\Environment($di, $loader);
 
         $this->twig->addExtension(new Twig\CoreExtension());
-        $this->registryFunctions($view, $di, $userFunctions);
+        $this->registryFunctions($view, $di);
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/View/Engine/Twig.php]: Why?

**Comment by endeveit** [comment]
Done in #348

**Pull request closed without merging**